src/charm.py

```diff
@@ -91,19 +120,13 @@ def _on_loki_push_api_endpoint_departed(self, event) -> None:
         """Event handler for the loki departed."""
         self._update_config(event)
 
-    def on_pebble_ready(self, event: EventBase) -> None:
+    def on_pebble_ready(self, _: EventBase) -> None:
         """Event handler for the pebble ready event.
 
         Args:
             event: The event object of the pebble ready event
         """
-        container = event.workload
-        if not self._stored.agent_config:
-            config = self._config_file(event)
-            container.push(CONFIG_PATH, config)
-            self._stored.agent_config = config
-        else:
-            container.push(CONFIG_PATH, self._stored.agent_config)
+        self._container.push(CONFIG_PATH, yaml.dump(self._config_file()), make_dirs=True)
```

Comment:
If we get another PR, a guard wouldn't hurt here. The container is _almost certain_ to be contacted right after `pebble_ready`, but better to be safe...